mlrun/utils/notifications/condition_evaluator.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import multiprocessing
+
+import mlrun.model
+from mlrun.utils import logger
+
+from .utils import sanitize_notification
+
+
+def evaluate_notification_condition_in_separate_process(
+    run: mlrun.model.RunObject, notification: mlrun.model.Notification, timeout: int = 5
+):
+    receiver, sender = multiprocessing.Pipe()
+    p = multiprocessing.Process(
+        target=_evaluate_notification_condition_wrapper,
+        args=(sender, run, notification),
+    )
+    p.start()
+    if receiver.poll(timeout):
+        result = receiver.recv()
+        p.join()
+        return result
+    else:
+        p.kill()
+        logger.warning(
+            f"Condition evaluation timed out after {timeout} seconds. Ignoring condition",
+            notification=sanitize_notification(notification),
+        )
+        return True
+
+
+def _evaluate_notification_condition_wrapper(
+    connection, run: mlrun.model.RunObject, notification: mlrun.model.Notification
+):
+    connection.send(_evaluate_notification_condition(run, notification))
+    return connection.close()
+
+
+def _evaluate_notification_condition(
+    run: mlrun.model.RunObject, notification: mlrun.model.Notification
+):
+    if not notification.condition:
+        return True
```

Comment:
I would move that to a step before you go through all the effort of creating a new process to basically do nothing... Should put this in `evaluate_notification_condition_in_separate_process`.